### ISSUE TITLE
单机多卡训练错误

### ISSUE THREAD

**Issue by Geministudents** [ymcui/Chinese-LLaMA-Alpaca#773]
### 提交前必须检查以下项目

- [X] 请确保使用的是仓库最新代码（git pull），一些问题已被解决和修复。
- [X] 由于相关依赖频繁更新，请确保按照[Wiki](https://github.com/ymcui/Chinese-LLaMA-Alpaca/wiki)中的相关步骤执行
- [X] 我已阅读[FAQ章节](https://github.com/ymcui/Chinese-LLaMA-Alpaca/wiki/常见问题)并且已在Issue中对问题进行了搜索，没有找到相似问题和解决方案
- [X] 第三方插件问题：例如[llama.cpp](https://github.com/ggerganov/llama.cpp)、[text-generation-webui](https://github.com/oobabooga/text-generation-webui)、[LlamaChat](https://github.com/alexrozanski/LlamaChat)等，同时建议到对应的项目中查找解决方案
- [X] 模型正确性检查：务必检查模型的[SHA256.md](https://github.com/ymcui/Chinese-LLaMA-Alpaca/blob/main/SHA256.md)，模型不对的情况下无法保证效果和正常运行

### 问题类型

模型训练与精调

### 基础模型

Alpaca-13B

### 操作系统

Linux

### 详细描述问题

CUDA_VISIBLE_DEVICES=0,1 torchrun --nproc_per_node 2 --master_port 25001 run_clm_sft_with_peft.py \
    --deepspeed ${deepspeed_config_file} \
    --model_name_or_path ${pretrained_model} \
    --tokenizer_name_or_path ${chinese_tokenizer_path} \
    --dataset_dir ${dataset_dir} \
    --validation_split_percentage 0.001 \
    --per_device_train_batch_size ${per_device_train_batch_size} \
    --per_device_eval_batch_size ${per_device_eval_batch_size} \
    --do_train \
    --do_eval \
    --seed 3407 \
    --fp16 \
    --num_train_epochs 1 \
    --lr_scheduler_type cosine \
    --learning_rate ${lr} \
    --warmup_ratio 0.03 \
    --weight_decay 0 \
    --logging_strategy steps \
    --logging_steps 10 \
    --save_strategy steps \
    --save_total_limit 1 \
    --evaluation_strategy steps \
    --eval_steps 100 \
    --save_steps 200 \
    --gradient_accumulation_steps ${gradient_accumulation_steps} \
    --preprocessing_num_workers 8 \
    --max_seq_length 512 \
    --output_dir ${output_dir} \
    --overwrite_output_dir \
    --ddp_timeout 30000 \
    --logging_first_step True \
    --torch_dtype float16 \
    --peft_path ${peft_model} \
    --validation_file ${validation_file} \
    --ddp_find_unused_parameters False \

### 依赖情况（代码类问题务必提供）

Package                  Version
------------------------ ------------
accelerate               0.21.0
aiohttp                  3.8.5
aiosignal                1.3.1
async-timeout            4.0.2
attrs                    23.1.0
beautifulsoup4           4.12.2
certifi                  2023.5.7
charset-normalizer       3.2.0
cmake                    3.26.4
datasets                 2.13.1
deepspeed                0.10.0
dill                     0.3.6
filelock                 3.12.2
frozenlist               1.4.0
fsspec                   2023.6.0
google                   3.0.0
hjson                    3.1.0
huggingface-hub          0.16.4
idna                     3.4
Jinja2                   3.1.2
lit                      16.0.6
MarkupSafe               2.1.3
mpmath                   1.3.0
multidict                6.0.4
multiprocess             0.70.14
networkx                 3.1
ninja                    1.11.1
numpy                    1.24.4
nvidia-cublas-cu11       11.10.3.66
nvidia-cuda-cupti-cu11   11.7.101
nvidia-cuda-nvrtc-cu11   11.7.99
nvidia-cuda-runtime-cu11 11.7.99
nvidia-cudnn-cu11        8.5.0.96
nvidia-cufft-cu11        10.9.0.58
nvidia-curand-cu11       10.2.10.91
nvidia-cusolver-cu11     11.4.0.1
nvidia-cusparse-cu11     11.7.4.91
nvidia-nccl-cu11         2.14.3
nvidia-nvtx-cu11         11.7.91
packaging                23.1
pandas                   2.0.3
peft                     0.3.0.dev0
pip                      23.1.2
protobuf                 4.23.4
psutil                   5.9.5
py-cpuinfo               9.0.0
pyarrow                  12.0.1
pydantic                 1.10.11
python-dateutil          2.8.2
pytz                     2023.3
PyYAML                   6.0.1
regex                    2023.6.3
requests                 2.31.0
safetensors              0.3.1
sentencepiece            0.1.97
setuptools               67.7.2
six                      1.16.0
soupsieve                2.4.1
sympy                    1.12
tokenizers               0.13.3
torch                    1.13.1+cu117
tqdm                     4.65.0
transformers             4.30.0
triton                   2.0.0
typing_extensions        4.7.1
tzdata                   2023.3
urllib3                  2.0.3
wheel                    0.40.0
xxhash                   3.2.0
yarl                     1.9.2

### 运行日志或截图

![image](https://github.com/ymcui/Chinese-LLaMA-Alpaca/assets/45708697/78c4609e-fa3d-41e9-bcc2-07e1df22c2c7)
已经换了很多版本的python和torch、transformers的包了，但是依旧存在这种问题，两块A40（45G）的GPU，当分配进程，两块GPU的内存都占据一半的时候，就会发生一块GPU的进程中断，求解，希望各位能提出宝贵意见，谢谢各位

**Comment by Geministudents** [comment]
补充一点：单机单卡是可以的....不知道为什么多卡就会出现问题

**Comment by Zhou-Zoey** [comment]
你好，请问解决了吗，我遇到了一模一样的问题

**Comment by Geministudents** [comment]
> 

还未解决，如果是在docker中，可以尝试在本地跑一跑，看别人说可能和容器有关

**Comment by LMXKO** [comment]
你是全参微调，不是lora是吗

**Comment by Zhou-Zoey** [comment]
> 

是仅lora，而且我发现单机单卡也会出现这个问题？

**Comment by Zhou-Zoey** [comment]
如果在docker中可以尝试增大共享内存，参考这个 https://github.com/microsoft/DeepSpeed/issues/2897
